### PR TITLE
Add ga_sessions_v1 table & view

### PR DESCRIPTION
**Pull request by fbertsch**
This table aggregates session-level data from GA.

Checklist for reviewer:

- [ ] Commits should reference a bug or github issue, if relevant (if a bug is referenced, the pull request should include the bug number in the title).
- [ ] If the PR comes from a fork, trigger integration CI tests by running the [Push to upstream workflow](https://github.com/mozilla/bigquery-etl/actions/workflows/push-to-upstream.yml) and provide the `<username>:<branch>` of the fork as parameter. The parameter will also show up
in the logs of the `manual-trigger-required-for-fork` CI task together with more detailed instructions.
- [ ] If adding a new field to a query, ensure that the schema and dependent downstream schemas have been updated.
- [ ] When adding a new derived dataset, ensure that data is not available already (fully or partially) and recommend extending an existing dataset in favor of creating new ones. Data can be available in the [bigquery-etl repository](https://github.com/mozilla/bigquery-etl), [looker-hub](https://github.com/mozilla/looker-hub) or in [looker-spoke-default](https://github.com/mozilla/looker-spoke-default/tree/e1315853507fc1ac6e78d252d53dc8df5f5f322b).

For modifications to schemas in restricted namespaces (see [`CODEOWNERS`](https://github.com/mozilla/bigquery-etl/blob/main/CODEOWNERS)):
- [ ] Follow the [change control procedure](https://docs.google.com/document/d/1TTJi4ht7NuzX6BPG_KTr6omaZg70cEpxe9xlpfnHj9k/edit#heading=h.ttegrcfy18ck)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/DENG-1963)

### CODE REVIEW

**Comment by fbertsch** [comment]
Depends on https://github.com/mozilla/bigquery-etl/pull/4556/files

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/dataset_metadata.yaml
Comment (whd) [inline sql/moz-fx-data-shared-prod/mozilla_org_derived/dataset_metadata.yaml:11]: Table ACLs are additive to Dataset ACLs. Therefore, in order to restrict access to https://github.com/mozilla/bigquery-etl/pull/4557/files#diff-7d19dc15354cd1c33edcdb1e504d4617804c730316620813fcb43e800ccd3d4aR22 within this dataset the dataset ACL here has to be something more restrictive than `workgroup:mozilla-confidential` (like `[]`), and all other tables in the dataset that are `workgroup:mozilla-confidential` would need to be annotated specifically (or potentially automatically via `default_table_workgroup_access`).

Additionally the dataset base acl here being `derived` grants implicit read/write access to DE. That needs to be `derived_restricted` if it's not considered permissible for all of DE to have read/write access to any workgroup-confidential data in the dataset. In that configuration if anyone apart from Airflow needs direct write access to the dataset that would need to be granted here as well (https://mozilla-hub.atlassian.net/browse/DSRE-1023).

Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/dataset_metadata.yaml:11]: Many thanks, indeed we'll lock down the stub attribution logs in this follow-up PR: https://github.com/mozilla/bigquery-etl/pull/4557

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (ANich) [inline sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:82]: Would it be acceptable to process data once it's "settled" 3 days later?

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (ANich) [inline sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:89]: This part broke my brain a little, how will `@session_date` be set? Isn't this getting data from the future? See above question on getting data once it's settled.

**Review by ANich — COMMENTED** [review]
Main question is about handling `@session_date` - is the case for updating past sessions a future concern?

File: sql/moz-fx-data-shared-prod/mozilla_org/ga_sessions/view.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org/ga_sessions/view.sql:11]: Honestly, just because we don't have to. Because it's a 1:1 scalar transform I generally prefer to keep it as superficial as possible (just in case we want to investigate the pre-transformed data in some way).

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/metadata.yaml
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/metadata.yaml:10]: You're right, I realized the same when I was looking at the generated DAG earlier - thank you!

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:2]: Compatible as in - we are able to get the same data. Usually it means "this field is available in some form in GA4", but the only one where that is not the case is `mobileDeviceInfo` (in GA4, that field is split into its components, like manufacturer and model). My thought was we could recreate the field in the view for backwards compatibility.

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:75]: Definitely. This is probably worth doing alongside the GA4 migration

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:82]: That's probably waiting too long. I would prefer to have this updated daily (and it's usually a large majority of the data that arrives same-day).

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (fbertsch) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:89]: I'm writing an Airflow job that will process (today-1, today-2, and today-3). That will incrementally re-process the past 3 days, in which case they will look into the next day to ensure they have the full session information.

Comment (ANich) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:89]: This seems pretty complex. Won't any tables downstream of this that don't have late-arriving data have to do the same then? Or risk having incorrect data

**Review by ANich — APPROVED** [review]
The reprocessing part seems a little complex. Waiting 3-days for complete data rather than dealing with incomplete data downstream and in dashboards seems worth the tradeoff - ✅ to unblock

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (gkabbz) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:75]: I don't know if we have much option for the ga_sessions_*. GA basically drops this data in this form in our warehouse. We don't do much to it. Looks like they are still doing it even for the GA4 datasets

File: sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql
Comment (gkabbz) [reply sql/moz-fx-data-shared-prod/mozilla_org_derived/ga_sessions_v1/query.sql:82]: I can see the utility of being able to see the campaign immediately but I'd be worried about changing numbers and how peeps might feel if reports shift on them.

**Comment by fbertsch** [comment]
@ANich @gkabbz many thanks for the review and thoughts here.

In terms of the complication of handling late-arriving data vs. waiting, I strongly believe that we will decrease trust in the platform if we are not able to handle this in a manner similar to other tools. In the GA dashboards, for example, they don't wait for data to settle before displaying. For some ad-hoc & fast-turnaround analyses, this is an acceptable tradeoff.

For other datasets and metrics (dare I say, silver-level?), we may want to wait to report them until the data has settled. I would like to leave that to downstream datasets/metrics, however, than force it on them here.

**Pull request merged**
